### PR TITLE
Extend facet marking to 1D meshes

### PR DESCRIPTION
**Pull request by francesco-ballarin**
After discussion with @ksagiyam. Please double check, the test passes but I am not fully sure of what I am doing ;)

### CODE REVIEW

File: firedrake/mesh.py [hunk 1]
```diff
@@ -1259,11 +1260,15 @@ def mark_entities(self, tf, label_name, label_value):
         if elem.family() in {"Discontinuous Lagrange", "DQ"} and elem.degree() == 0:
             # cells
             height = 0
-        elif elem.family() == "HDiv Trace" and elem.degree() == 0:
+        elif (
+            (elem.family() == "HDiv Trace" and elem.degree() == 0 and self.cell_dimension() > 1)
+                or
+            (elem.family() == "Lagrange" and elem.degree() == 1 and self.cell_dimension() == 1)
+        ):
```
Comment (ksagiyam) [inline firedrake/mesh.py:1267]: Lint will fail, so we want something like:
```
        elif self.cell_dimension() == 1 and elem.family() == "Lagrange" and elem.degree() == 1 or \
                self.cell_dimension() > 1 and elem.family() == "HDiv Trace" and elem.degree() == 0:
```

File: tests/regression/test_mark_entities.py [hunk 2]
```diff
@@ -90,3 +90,16 @@ def test_mark_entities_mesh_mark_entities():
     label = plex.getLabel(label_name)
     assert label.getStratumIS(label_value).getSize() == 2
     assert all(label.getStratumIS(label_value).getIndices() == [20, 30])
+
```
Comment (ksagiyam) [inline tests/regression/test_mark_entities.py:93]: Want double blank lines.

**Review by ksagiyam — CHANGES_REQUESTED** [review]
Thanks! I requested some minor changes, but otherwise looks good to me.

File: firedrake/mesh.py (hunk 1 above)
Comment (ksagiyam) [reply firedrake/mesh.py:1267]: and similar change in `RelabeledMesh`.

**Comment by francesco-ballarin** [comment]
Not sure why it still says "changes requested", I believe I addressed all of them...

**Comment by wence-** [comment]
> Not sure why it still says "changes requested", I believe I addressed all of them...

@ksagiyam has to agree that you addressed them as requester :)

**Review by ksagiyam — APPROVED** [review]

**Comment by ksagiyam** [comment]
Done!

**Comment by francesco-ballarin** [comment]
Thanks @ksagiyam!

**Pull request merged**